### PR TITLE
just some help with the Readme documentation

### PR DESCRIPTION
**Pull request by alfadormx**
Hello

I humbly propose the following modifications mainly done in your Readme file. Their objective is to help onboard new developers into your SDK.

I can't wait to hear back from you and your feedback.

Best Regards
alFadorMX

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,22 +51,71 @@ public class MyTouchPortalPlugin extends TouchPortalPlugin {
...
+    /**
+     * Called when a Settings Message is received
+     */
+    public void onSettings(TPSettingsMessage tpSettingsMessage) { }
+}
+```
+
+## Development and Interaction
+
+- The SDK will automatically callback your action methods if they only contain a `@Data` annotated parameter
```
Comment (ChristopheCVB) [inline README.md:94]: - The SDK will automatically callback your action methods if they only contain `@Data` annotated parameters

File: README.md [hunk 2]
```diff
@@ -74,15 +129,46 @@ public class MyTouchPortalPlugin extends TouchPortalPlugin {
...
+public class MyTouchPortalPlugin extends TouchPortalPlugin implements TouchPortalPlugin.TouchPortalPluginListener {
+    // ...
+    public void onInfo(TPInfoMessage tpInfoMessage) {
+        // tpInfoMessage will contain the initial settings stored by TP
+        // continue plugin initialization
+    }
+}
+```
+
+- Finally, send notifications back to TouchPortal once you update your states
```
Comment (ChristopheCVB) [inline README.md:149]: - Finally, send messages back to Touch Portal when you want to update your states

File: README.md [hunk 3]
```diff
@@ -74,15 +129,46 @@ public class MyTouchPortalPlugin extends TouchPortalPlugin {
...
+public class MyTouchPortalPlugin extends TouchPortalPlugin implements TouchPortalPlugin.TouchPortalPluginListener {
+    // ...
+    @State(defaultValue = "Default Value", categoryId = "SecondCategory")
+    private String customStateText;
+  
+    @Action(description = "Long Description of Dummy Action with Data Text", format = "Set text to {$text$}", categoryId = "BaseCategory")
+    private void actionWithText(@Data String text) {
+      // ... do something then update state
+      this.customStateText = "new state value";
+      this.sendStateUpdate(MyTouchPortalPlugin.BaseCategory.States.CustomStateText.ID, this.customStateText, true);
```
Comment (ChristopheCVB) [inline README.md:161]: `MyTouchPortalPluginConstants.BaseCategory`...

File: README.md [hunk 4]
```diff
@@ -74,15 +129,46 @@ public class MyTouchPortalPlugin extends TouchPortalPlugin {
...
+    }
+}
+```
+
+## Use 'annotations' to help you package your plugin
```
Comment (ChristopheCVB) [inline README.md:166]: ## Use  Annotations to describe your plugin and package it

File: README.md [hunk 5]
```diff
@@ -74,15 +129,46 @@ public class MyTouchPortalPlugin extends TouchPortalPlugin {
...
+
+- The provided Annotations help you in the automatic generation of the `entry.tp` file (necessary for packaging and deployment of your plugin)
+- Current supported annotations include: Action, Category, Data, Event, Plugin, Setting and State
```
Comment (ChristopheCVB) [inline README.md:169]: - Current supported annotations include: Plugin, Category, Action, Data, State, Event and Setting

**Review by ChristopheCVB — CHANGES_REQUESTED** [review]
Hi there!
Thank you for the PR!

Just some tiny bits of wording adjustments and we'll be good to go ;)

**Comment by alfadormx** [comment]
Thanks for the review. I updated the Readme following the proposed wording changes. 
I hope my following PR might be more significate than this one.

**Review by ChristopheCVB — APPROVED** [review]
👍

**Pull request merged**